packages/queued-request-controller/src/QueuedRequestMiddleware.ts

```diff
@@ -92,20 +92,38 @@ export const createQueuedRequestMiddleware = ({
       await messenger.call(
         QueuedRequestControllerActionTypes.enqueueRequest,
         async () => {
-          if (req.method === 'wallet_switchEthereumChain') {
+          if (
+            req.method === 'wallet_switchEthereumChain' ||
+            req.method === 'wallet_addEthereumChain'
+          ) {
             return next();
           }
 
-          const isBuiltIn = isNetworkType(networkClientIdForRequest);
-          const networkConfigurationForRequest = messenger.call(
+          const networkClientConfigurationForRequest = messenger.call(
             'NetworkController:getNetworkClientById',
             networkClientIdForRequest,
           ).configuration;
 
-          const currentProviderConfig = messenger.call(
+          const networkControllerState = messenger.call(
             'NetworkController:getState',
-          ).providerConfig;
+          );
+
+          const isBuiltIn = isNetworkType(networkClientIdForRequest);
+          let networkConfigurationForRequest;
+          if (!isBuiltIn) {
+            networkConfigurationForRequest =
+              networkControllerState.networkConfigurations[
+                networkClientIdForRequest
+              ];
+          } else {
+            // if its a built in
+            // Ideally we should be using only networkConfigurations, and networkClientIds &
+            // networkConfiguration.id should be the same thing.
+            networkConfigurationForRequest =
+              networkClientConfigurationForRequest;
```

Comment:
should this be `networkClientIdForRequest`?
```suggestion
              networkClientIdForRequest;
```

Reply:
hehe alot of similar looking variables. `networkClientConfigurationForRequest` is not unset on line 112, `networkConfigurationForRequest` the value getting set here is unset on line 112

Reply:
`networkClientConfigurationForRequest` is set on line 102